### PR TITLE
bot answers to /bonus with hello + username

### PR DESCRIPTION
**Pull request by Veetiknen**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import pkg from '@slack/bolt';
```
Comment (ujoni) [inline src/app.js]: You can import the `App` directly from the bolt:
```suggestion
import { App } from '@slack/bolt';
```
This removes the need to destructure the `pkg` later (`const { App } = pkg;`)

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+app.command('/bonus', async ({ command, ack, respond }) => {
+  await ack(); // Acknowledge the command
+  await respond(`Hello, <@${command.user_id}>! 👋`);
+});
+
+(async () => {
+  await app.start(); 
+  console.log('⚡️ Bolt app is running in Socket Mode!');
+})();
```
Comment (ujoni) [inline src/app.js]: While this gets around the "no top-level `await`" restriction, self-executing methods are a little ugly.
You could:
- Use the Promise [`.then()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then) API:
  ```typescript
  app.start().then(() => {
    console.log('⚡️ Bolt app is running in Socket Mode!');
  });
  ```
- Upgrade to Node 22 which allows top-level `await` (IIRC)

File: package.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "node src/app.js"
+  },
+  "author": "",
+  "dependencies": {
+    "@slack/bolt": "^4.0.1",
+    "@slack/rtm-api": "^7.0.1",
+    "@slack/web-api": "^7.7.0",
+    "body-parser": "^1.20.3",
```
Comment (ujoni) [inline package.json]: This library does not seem to be in use, so you could just chuck it.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    "dotenv": "^16.4.5"
```
Comment (ujoni) [inline package.json:17]: Move `dotenv` to `devDependencies` in the `package.json`.

It is a good practice to separate the app's dependencies from those used solely for its development. It is then easier to see which packages are deployed with the application code and which are just local stuff.

**Review by ujoni — CHANGES_REQUESTED** [review]
Few comments to ponder upon.

Good stuff! 👍🏻

File: package.json (hunk 3 above)
Comment (Veetiknen) [reply package.json]: fixed

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,3 @@
+node_modules
+.env
+tsconfig.json
```
Comment (ujoni) [inline .gitignore]: The `tsconfig.json` should not be in .gitignore - it is a shared configuration file that is needed by all developers interacting with the repository

File: src/app.js
Comment (ujoni) [inline src/app.js:1]: Define an out dir in the tscofig for the build artifacts, such as "lib" or "build" and add it to the .gitignore.
The TS compiler creates these files during the transpilation/build step and generated files are (usually) left out of the repository.

**Review by ujoni — CHANGES_REQUESTED** [review]
Two smol things!

File: .gitignore (hunk 5 above)
Comment (Veetiknen) [reply .gitignore]: fixed

File: src/app.js
Comment (Veetiknen) [reply src/app.js:1]: fixed (hopefully)

**Review by ujoni — DISMISSED** [review]

File: README.md [hunk 6]
```diff
@@ -1,2 +1,43 @@
 # slackbot-sharepoint-excel-reader
 A slackbot which reads an excel file from sharepoint and returns a specific column from it from a specific row according to the calling users slack handle
+
+> Uses `NodeJS` 20 and `npm` 10
+
+## Creation of the bot 
+### Setup bot 
+1. [Go to Slack API your apps](resources/docs/internal/aws_account_creation.md)
```
Comment (lockmhe) [inline README.md]: Is this link here correct?

**Review by lockmhe — CHANGES_REQUESTED** [review]

**Review by lockmhe — APPROVED** [review]

**Pull request merged**